contrib/autostyle.lua

```diff
@@ -0,0 +1,130 @@
+
+  -- check they all exist (correct syntax)
+  if (not tag) then
+	  darktable.print("EXIF TAG not found in " .. darktable.preferences.read("autostyle","exif_tag","string"))
+	  return
+  end
+  if (not value) then
+	  darktable.print("value to match not found in " .. darktable.preferences.read("autostyle","exif_tag","string"))
+	  return
+  end
+  if (not style_name) then
+	  darktable.print("style name not found in " .. darktable.preferences.read("autostyle","exif_tag","string"))
+	  return
+  end
+
+  -- First find the style (we have its name)
+  local styles= darktable.styles
+  local style
+  for _,s in ipairs(styles) do
+	  if s.name == style_name then
+		  style=s
+	  end
+  end
+  if (not style) then
+	  darktable.print("style not found for autostyle: " .. style_name)
+  end
+
+  -- Apply the style to image, if it is tagged
+  local auto_dr_attr= exiftool_attribute(image.path .. '/' .. image.filename,tag);
+  if auto_dr_attr==value then
+--	  darktable.print("Image " .. image.filename .. ": autostyle automatically applied " .. darktable.preferences.read("autostyle","exif_tag","string") )
+	  darktable.styles.apply(style,image)
+--  else
+--	  darktable.print("Image " .. image.filename .. ": autostyle not applied, exif tag " .. darktable.preferences.read("autostyle","exif_tag","string")  .. " not matched: " .. auto_dr_attr)
+  end
+end 
+
+
+function autostyle_apply( shortcut)
+  local images = darktable.gui.action_images
+  for _,image in pairs(images) do
+    autostyle_apply_one_image(image)
+  end
+end
+
+-- Retrieve the attribute through exiftool
+function exiftool_attribute(path,attr)
+  local cmd="exiftool -" .. attr .. " '" ..path.. "'";
+  local exifresult=get_stdout(cmd)
+  local attribute= assert(string.match(exifresult,": (.*)"), "Could not find the attribute " .. attr .. " using the command: <" .. cmd .. ">")
+  return attribute
+end
+
+-- run command and retrieve stdout
+function get_stdout(cmd)
+  -- Open the command, for reading
+  local fd = assert(io.popen(cmd, 'r'))
+  -- yield to other lua threads until data is ready to be read
+  coroutine.yield("FILE_READABLE",fd)
+  -- slurp the whole file
+  local data = assert(fd:read('*a'))
+
+  fd:close()
+  -- Replace carriage returns and linefeeds with spaces
+  data = string.gsub(data, '[\n\r]+', ' ')
+  -- Remove spaces at the beginning
+  data = string.gsub(data, '^%s+', '')
+  -- Remove spaces at the end
+  data = string.gsub(data, '%s+$', '')
+  return data
+end
+
+-- Registering events
+darktable.register_event("shortcut",autostyle_apply,
+       "Apply your chosen style from exiftool tags")
+
+darktable.preferences.register("autostyle","exif_tag","string","Autostyle: EXIF_tag=value=>style","apply a style automatically if an EXIF_tag matches value. Find the tag with exiftool","")
+
+darktable.register_event("post-import-image",autostyle_apply_one_image)
```

Comment:
this won't work. when called on the event, autostyle_apply_one_image will get the event name as its first parameter and the image as the second

so it will have "post-import-image" (the string) as its "image" parameter, leading to errors

please fix